client/js/s3/request-signer.js

```diff
@@ -265,7 +268,9 @@ qq.s3.RequestSigner = function(o) {
             pendingSignatureData = pendingSignatures[id],
             promise = pendingSignatureData.promise,
             signatureConstructor = pendingSignatureData.signatureConstructor,
-            errorMessage, response;
+            oldCredentials = Object.assign({}, credentialsProvider.get()),
```

Comment:
This was the ideal solution I was hoping existed, but in my cursory glance at the qq util functions I must not have seen this one.  Will update this later.